app/models/user.rb

```diff
@@ -85,7 +85,7 @@ class User < ApplicationRecord
   scope :deactivated, -> { where(active: false) }
   scope :active_status, -> { where(active: true).where.not(last_sign_in_at: nil) }
   scope :visible, -> { where(discarded_at: nil) }
-  scope :own_and_managing_org_users, ->(organisation) { where(organisation: organisation.child_organisations + [organisation]) }
+  scope :visible_to_user, ->(user) { user.support? ? visible : visible.where(organisation: user.organisation.child_organisations + [user.organisation]) }
```

Comment:
I've used the visible scope to filter out discarded users, then applied further filters based on the user's role and organisation.

Reply:
Does this need to be separate from `visible`? (I can't remember exactly where we're using what, but can't think of anywhere we'd need just "not discarded" in a non-support context)

Reply:
I've added another commit which aims to create a single visible scope. It requires access to the current user logged in to determine if they're a support user or not. In order to do so within the ApplicationController I set the current user in a thread local variable (`current_user`) and added a before action to set the user for each request. Within the User model we add a method `self.current` to grab the `:current_user` from the thread and use it within the scope to determine if the user is a support user or not. Could you have a look and comment on whether this approach is okay?